blockers.md

```diff
@@ -0,0 +1,24 @@
+# Blockers
+
+Issues with IPFS that are limiting package manager adoption today
+
+## File system based package managers
+
+### Requires 2x disk space for mirroring
+
+Filestore expects files to be immutable once added, so rsyncing updates to existing files [causes errors](https://github.com/protocol/package-managers/issues/18#issuecomment-471365124), one workaround is to not use the filestore but that requires copying the entire mirror directory into `.ipfs`.
+
+### No easy way to directly add a directory to MFS with go-ipfs
+
+Adding a directory of files to MFS means calling out to `ipfs files write` for every file, ideally there should be one command to write a directory of files to MFS.
```

Comment:
For posterity, I'll mention that I'm currently working on a variant of `go-ipfs` that has writable access to the Files(MFS) and Name(IPNS) APIs, through the FUSE API.

While not fully viable today, I would like it to be capable of hosting a living/mutable package repo in the same way a traditional filesystem can (to the specifications our users expect).

Some issues I encountered while writing this have been resolved already, but there's a handful that are not simple fixes. Since this demand is pretty broad, I look forward to ~bugging~ collaborating with a bunch of us here to try and harmonize some of the next iterations of these APIs and formats to enable this kind of use case.

@aschmahmann's current work on IPNS will also likely have a bearing on this. I have an implementation to expose IPNS keys as writable files and directories through MFS which may be useful for package managers.  It's basically an abstraction on the existing common pattern (modify a root via `ipfs files` -> stat changes -> `name publish`)